sale_coupon_domain_product_discount/models/coupon_reward.py

```diff
@@ -1,22 +1,79 @@
 # Copyright 2022 Ooops404
+# Copyright 2022 Tecnativa - David Vidal
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import _, api, fields, models
+from odoo.tools.safe_eval import safe_eval
 
-from odoo import _, fields, models
 
+class SaleCouponReward(models.Model):
+    _inherit = "sale.coupon.reward"
+    # Now the main field is a non stored field. We'll be storing former records for
+    # regular behavior a special sibling stored field
+    discount_specific_product_ids = fields.Many2many(
+        compute="_compute_discount_specific_product_ids",
+        inverse="_inverse_discount_specific_product_ids",
+        search="_search_discount_specific_product_ids",
+        readonly=False,
+        store=False,
+    )
+    # This is where we'll store our real products for regular promos. Indeed, we're
+    # taking the former relation in order to keep former data. This saves us
+    # supercomplex init/uninstall hooks +1 for the ORM
+    discount_specific_product_ids_stored = fields.Many2many(
```

Comment:
https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#38fields

I think no need add additional string after suffix `_ids`. Maybe `stored_discount_specific_product_ids`

Reply:
@chienandalu what do you think ?

Reply:
Perfect :+1: Changed